File: Java Project/Iteration_2/src/ElectiveCourse.java
package Iteration_2.src;


public class ElectiveCourse {

    private int quota;
    private Schedule schedule;
    private String electiveType;

    public ElectiveCourse(String courseCode, String courseName, int credit, int quota, Schedule schedule, String electiveType) {
        this.quota = quota;
        this.schedule = schedule;
        this.electiveType = electiveType;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public ElectiveCourse(int quota, Schedule schedule, String electiveType) {
        this.quota = quota;
        this.schedule = schedule;
        this.electiveType = electiveType;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public int getQuota() {
        return quota;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public void setQuota(int quota) {
        this.quota = quota;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public Schedule getSchedule() {
        return schedule;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public void setSchedule(Schedule schedule) {
        this.schedule = schedule;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public String getElectiveType() {
        return electiveType;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }

    public void setElectiveType(String electiveType) {
        this.electiveType = electiveType;
        Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        LoggingAgent.LoggerMethod();
    }
}

File: Python/src/ElectiveCourse.py
class ElectiveCourse:
    def __init__(self, course_code, course_name, credit, quota, schedule, elective_type):
        self.__quota = quota
        self.__schedule = schedule
        self.__elective_type = elective_type

    def __init__(self, quota, schedule, elective_type):
        self.__quota = quota
        self.__schedule = schedule
        self.__elective_type = elective_type

    def get_quota(self):
        return self.__quota

    def set_quota(self, quota):
        self.__quota = quota

    def get_schedule(self):
        return self.__schedule

    def set_schedule(self, schedule):
        self.__schedule = schedule

    def get_elective_type(self):
        return self.__elective_type

    def set_elective_type(self, elective_type):
        self.__elective_type = elective_type
